src/System Application/App/File System/src/Account/FileAccountImpl.Codeunit.al

```diff
@@ -0,0 +1,247 @@
+            exit;
+
+        // Get the current default account to track if it was deleted
+        FileScenario.GetDefaultFileAccount(CurrentDefaultFileAccount);
+
+        // Delete all selected acounts
+        repeat
+            // Check to validate that the connector is still installed
+            // The connector could have been uninstalled by another user/session
+            if IsValidConnector(FileAccountsToDelete.Connector) then begin
+                FileSystemConnector := FileAccountsToDelete.Connector;
+                FileSystemConnector.DeleteAccount(FileAccountsToDelete."Account Id");
+            end;
+        until FileAccountsToDelete.Next() = 0;
+
+        DefaultAccountDeletion(CurrentDefaultFileAccount."Account Id", CurrentDefaultFileAccount.Connector);
+    end;
+
+    local procedure DefaultAccountDeletion(CurrentDefaultAccountId: Guid; Connector: Enum "File System Connector")
+    var
+        AllFileAccounts: Record "File Account";
+        NewDefaultFileAccount: Record "File Account";
+        FileScenario: Codeunit "File Scenario";
+    begin
+        GetAllAccounts(false, AllFileAccounts);
+
+        if AllFileAccounts.IsEmpty() then
+            exit; //All of the accounts were deleted, nothing to do
+
+        if AllFileAccounts.Get(CurrentDefaultAccountId, Connector) then
+            exit; // The default account was not deleted or it never existed
+
+        // In case there's only one account, set it as default
+        if AllFileAccounts.Count() = 1 then begin
+            MakeDefault(AllFileAccounts);
+            exit;
+        end;
+
+        Commit();  // Commit the accounts deletion in order to prompt for new default account
+        if PromptNewDefaultAccountChoice(NewDefaultFileAccount) then
+            MakeDefault(NewDefaultFileAccount)
+        else
+            FileScenario.UnassignScenario(Enum::"File Scenario"::Default); // remove the default scenario as it is pointing to a non-existent account
+    end;
+
+    local procedure PromptNewDefaultAccountChoice(var NewDefaultFileAccount: Record "File Account"): Boolean
+    var
+        FileAccountsPage: Page "File Accounts";
+    begin
+        FileAccountsPage.LookupMode(true);
+        FileAccountsPage.EnableLookupMode();
+        FileAccountsPage.Caption(ChooseNewDefaultTxt);
+        if FileAccountsPage.RunModal() = Action::LookupOK then begin
+            FileAccountsPage.GetAccount(NewDefaultFileAccount);
+            exit(true);
+        end;
+
+        exit(false);
+    end;
+
+    local procedure ImportLogo(var FileAccount: Record "File Account"; Connector: Interface "File System Connector")
+    var
+        FileConnectorLogo: Record "File System Connector Logo";
+        TempBlob: Codeunit "Temp Blob";
+        Base64Convert: Codeunit "Base64 Convert";
+        ConnectorLogoBase64: Text;
+        OutStream: Outstream;
+        InStream: InStream;
+        ConnectorLogoDescriptionTxt: Label '%1 Logo', Locked = true;
```

Comment:
Label comment for %1

Reply:
Rule [AA0470](https://learn.microsoft.com/de-de/dynamics365/business-central/dev-itpro/developer/analyzers/codecop-aa0470?wt.mc_id=d365bc_inproduct_alextension#reason-for-the-rule) doesn't apply here. It is locked.